docs/source/en/optimization/coreml.mdx

```diff
@@ -0,0 +1,164 @@
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# How to run Stable Diffusion natively on Mac and iPhone
+
+There are two ways to run Stable Diffusion on Apple hardware:
+
+- Using the `diffusers` Python codebase and leveraging PyTorch's `mps` device. This is suitable for Macs running Apple Silicon chips (M1 or M2), and is explained in [the `mps` guide](mps).
+
+- Converting Stable Diffusion models to Core ML, the model format supported by Apple frameworks, and using the converted models in native apps. This works on all types of Apple hardware, including iPhone, iPad, M1/M2 Macs and even Intel Macs with enough GPU memory.
+
+## Stable Diffusion Core ML Checkpoints
+
+Core ML is the native ML model format for Apple devices. Because Stable Diffusion was trained using PyTorch, the model weights are available in PyTorch format. Therefore, we need to convert them to Core ML before we can use them inside native apps.
+
+Thankfully, Apple engineers developed [a conversion tool](https://github.com/apple/ml-stable-diffusion) based on `diffusers`. If you have a Mac, you can use the conversion scripts it provides to convert any Stable Diffusion model to Core ML.
+
+Before you convert a model, though, do take a moment to explore the Hugging Face Hub – chances are the model you're interested in is already available in Core ML format. For example, these official models are available in Core ML in the Apple organization:
+
+- Stable Diffusion v1.4: [converted](https://hf.co/apple/coreml-stable-diffusion-v1-4) [original](https://hf.co/CompVis/stable-diffusion-v1-4)
+- Stable Diffusion v1.5: [converted](https://hf.co/apple/coreml-stable-diffusion-v1-5) [original](https://hf.co/runwayml/stable-diffusion-v1-5)
+- Stable Diffusion v2 base: [converted](https://hf.co/apple/coreml-stable-diffusion-2-base) [original](https://huggingface.co/stabilityai/stable-diffusion-2-base)
+- Stable Diffusion v2.1 base: [converted](https://hf.co/apple/coreml-stable-diffusion-2-1-base) [original](https://huggingface.co/stabilityai/stable-diffusion-2-1-base)
+
+If you are interested in custom Dreamboothed or fine-tuned models, you can take a look at the [`coreml` community organization](https://huggingface.co/coreml) or use [this query to browse](https://huggingface.co/models?pipeline_tag=text-to-image&library=coreml&p=2&sort=likes) through all available models.
+
+If you can't find the model you're interested in, we recommend you follow the instructions in [Apple's repo](https://github.com/apple/ml-stable-diffusion) to perform the conversion yourself.
```

Comment:
I think it might be nice to also include the command to convert the model to CoreML just to show the user it only requires a single line of code/and it is pretty straightforward. If most users can execute this command and get the converted CoreML model without having to jump to another page, I think that's better for them, and its less disruptive to their workflow :)

```suggestion
If you can't find the model you're interested in, we recommend you follow the instructions in [Apple's repo](https://github.com/apple/ml-stable-diffusion#-converting-models-to-core-ml) to perform the conversion yourself.
```